addons/web/static/src/js/views/calendar/calendar_model.js

```diff
@@ -482,14 +483,41 @@ return AbstractModel.extend({
             })
             .then(function (events) {
                 self._parseServerData(events);
-                self.data.data = _.map(events, self._recordToCalendarEvent.bind(self));
+                // Fullcalendarjs sort the event by ID but it is not always the same as sorted by start date.
+                // _sortCustomId set an ID on the event that depend on the starting time of the events
+                const eventsNewIds = self._sortCustomId(events);
+                self.data.data = _.map(eventsNewIds, self._recordToCalendarEvent.bind(self));
                 return Promise.all([
                     self._loadColors(self.data, self.data.data),
                     self._loadRecordsToFilters(self.data, self.data.data)
                 ]);
             });
         });
     },
+        /**
+     * @private
+     * @param {any} events
+     * @returns {any} events with new property: fullcalendarId
+     */
+     _sortCustomId: function (events) {
+        // Sort events by their start_date
+        events.sort(function (a, b) {
+            a = a.start;
+            b = b.start;
+            if (a !== undefined && b !== undefined) {
+                return a.unix() - b.unix();
+            } else {
+                return 0;
+            }
+        });
+         let fcId = 1;
+         for (const key in events) {
```

Comment:
fullcalendar id won't be unique then, is that problematic ?

Reply:
Nope. Actually, you can have all id identical or even don't set it in _recordToCalendarEvent.
In that case, the first time you launch the page, the events are rendered in the expected order (by start date) but when you change the start date to modify the expected order, your events will be rendered in the same old order. You have to fully reload the page to see the modification.